### ISSUE TITLE
Style/RedundantReturn cop does not detect any offences

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12106]
## Actual behavior

I have the following code:
```ruby
# frozen_string_literal: true

module ApplicationHelper
  def my_method(condition)
    return 'active' if condition
  end
end
```

I run rubocop and don't get any offences.
```bash
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop app/helpers/application_helper.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## RuboCop version

```bash
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.55.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.18.0
  - rubocop-factory_bot 2.23.1
  - rubocop-performance 1.18.0
  - rubocop-rails 2.20.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.23.0
  - rubocop-thread_safety 0.5.1
```

**Comment by dvandersluis** [comment]
Your code is explicitly an example of a "good" use in the documentation: 

```ruby
# good
def test
  return something if something_else
end
```

It looks like it was intended to be implemented in the future but never happened. I'll take a look at it now.

**Comment by koic** [comment]
I had already opened #12108 as a false negative. Thank you.
